storage/series/series.go

```diff
@@ -412,6 +432,35 @@ func (s *dbSeries) Bootstrap(blocks block.DatabaseSeriesBlocks) error {
 	return multiErr.FinalError()
 }
 
+func (s *dbSeries) OnRetrieveBlock(
+	id ts.ID,
+	startTime time.Time,
+	segment ts.Segment,
+) {
+	s.Lock()
+	defer s.Unlock()
+
+	if !id.Equal(s.id) {
```

Comment:
is just this a sanity test?

Reply:
Correct yeah, potentially once upon a time it was possible to list yourself as the `OnRetrieveBlock(...)` and then by the time the ref was actually used you had been recycled and were in use for another series or just sitting in the object pool.

This isn't the case anymore since `databaseShard` is the one that gets called back and uses an RLock to make sure only a valid ref would ever get called with `OnRetrieveBlock(...)` though.